lib/install/vchlog/buffered_pipe.go

```diff
@@ -76,7 +87,26 @@ func (bp *BufferedPipe) Write(data []byte) (n int, err error) {
 func (bp *BufferedPipe) Close() (err error) {
 	bp.c.L.Lock()
 	defer bp.c.L.Unlock()
-	defer bp.c.Signal()
-	bp.closed = true
-	return nil
+	defer bp.c.Broadcast()
+
+	bp.writeClosed = true
+
+	if bp.readerReady { // flush only if there is a valid consumer
+		done := make(chan bool)
+		for bp.buffer.Len() > 0 {
+			go func() {
+				bp.c.Wait()
+				done <- true
+			}()
+			select {
+			case <-time.After(time.Minute * 30): // timeout if consumer inactive for 30 minutes
```

Comment:
I don't like timeouts - they just cause pain, but....
30 min seems like a long time given the amount of data we're trying to move and the expected duration of the operation. This blocks Close from returning - does that prevent the API request from completing?

I'm wondering if we should chose an equally horrific approach of creating the pipe with a context, saving that into the structure and then waiting for that to time out or be cancelled. At least then we're exposed to cancellation by way of the client http connection dropping.

Reply:
Yes, the timeout does prevent the API request from returning. The VCH will still be created, but the close hangs so the API hangs as well. Also I'm not sure what would that mean for the vic-machine service and other processes running on the service

I'm not sure if it makes sense for a pipe stream to have its caller context. Maybe it's vchlogger who needs to have the context and keep track of context timeouts?